substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoFeature.java

```diff
@@ -62,4 +86,56 @@ public void afterRegistration(AfterRegistrationAccess access) {
             }
         }
     }
+
+    @Override
+    @SuppressWarnings("try")
+    public void beforeImageWrite(BeforeImageWriteAccess access) {
+        Timer timer = TimerCollection.singleton().get(TimerCollection.Registry.DEBUG_INFO);
+        try (Timer.StopTimer t = timer.start()) {
+            ImageSingletons.add(SourceManager.class, new SourceManager());
+            var accessImpl = (FeatureImpl.BeforeImageWriteAccessImpl) access;
+            var image = accessImpl.getImage();
+            var debugContext = new DebugContext.Builder(HostedOptionValues.singleton(), new GraalDebugHandlersFactory(GraalAccess.getOriginalSnippetReflection())).build();
```

Comment:
Is the use of `var` encouraged in the GraalVM code base? 
If so, I will need to keep that in mind :)

(I am personally not a fan, but that's just my personal taste)

Reply:
Yes, if it's easy from the context to guess what type the local is or the types are rather trivial, it's fine to use `var`.
If the type in non-trivial and being explicit actually helps to understand the code better then not using var is recommended.